provider/src/main/java/com/synopsys/integration/alert/provider/blackduck/task/accumulator/BlackDuckNotificationRetriever.java

```diff
@@ -7,66 +7,116 @@
+        BlackDuckRequestBuilder requestBuilder = createNotificationRequestBuilder(dateRange, types);
+        NotificationPageRetriever notificationRetriever = pageDef -> retrievePageOfFilteredNotifications(requestBuilder, pageDef);
+
+        BlackDuckPageDefinition firstPageDef = new BlackDuckPageDefinition(DEFAULT_PAGE_SIZE, INITIAL_PAGE_OFFSET);
+        BlackDuckPageResponse<NotificationView> firstPage = notificationRetriever.apply(firstPageDef);
+
+        return new BlackDuckNotificationPage(firstPageDef.getOffset(), firstPageDef.getLimit(), firstPage.getTotalCount(), firstPage.getItems(), notificationRetriever);
+    }
+
+    private BlackDuckPageResponse<NotificationView> retrievePageOfFilteredNotifications(BlackDuckRequestBuilder requestBuilder, BlackDuckPageDefinition pageDefinition) throws IntegrationException {
+        return blackDuckApiClient.getPageResponse(requestBuilder, NotificationView.class, pageDefinition);
+    }
+
+    private BlackDuckRequestBuilder createNotificationRequestBuilder(DateRange dateRange, List<String> notificationTypesToInclude) {
+        SimpleDateFormat sdf = new SimpleDateFormat(RestConstants.JSON_DATE_FORMAT);
+        sdf.setTimeZone(TimeZone.getTimeZone("UTC"));
+
+        String startDateString = toDateString(sdf, dateRange.getStart());
+        String endDateString = toDateString(sdf, dateRange.getEnd());
+
+        BlackDuckRequestFilter notificationTypeFilter = BlackDuckRequestFilter.createFilterWithMultipleValues("notificationType", notificationTypesToInclude);
+        return blackDuckRequestFactory
+                   .createCommonGetRequestBuilder()
+                   .addQueryParameter("startDate", startDateString)
+                   .addQueryParameter("endDate", endDateString)
+                   .addBlackDuckFilter(notificationTypeFilter);
+    }
+
+    private String toDateString(SimpleDateFormat sdf, OffsetDateTime offsetDateTime) {
+        Date date = Date.from(offsetDateTime.toInstant());
+        return sdf.format(date);
     }
 
-    private List<NotificationView> retrieveFilteredNotifications(NotificationService notificationService, DateRange dateRange, List<String> notificationTypeNames) {
-        OffsetDateTime startDate = dateRange.getStart();
-        OffsetDateTime endDate = dateRange.getEnd();
-        logger.info("Accumulating Notifications Between {} and {} ", DateUtils.formatDate(startDate, RestConstants.JSON_DATE_FORMAT), DateUtils.formatDate(endDate, RestConstants.JSON_DATE_FORMAT));
-        try {
-            List<NotificationView> notificationViews = notificationService.getFilteredNotifications(Date.from(startDate.toInstant()), Date.from(endDate.toInstant()), notificationTypeNames);
-            logger.debug("Read Notification Count: {}", notificationViews.size());
-            return notificationViews;
-        } catch (Exception e) {
-            logger.error("Error reading notifications", e);
+    public static class BlackDuckNotificationPage {
```

Comment:
Can we use an abstract variation of this class for all of our paging? Not sure what we use elsewhere but looks like this would be useful in other places.